spy/util.py

```diff
@@ -118,17 +119,56 @@ def shortrepr(s: str, n: int) -> str:
         s = s[:n-2] + '...'
     return repr(s)
 
+def unbuffer_run(cmdline_s: Sequence[str]) -> subprocess.CompletedProcess:
+    """
+    Simulate the behavior of the unbuffer command from the expect package.
+    
+    Like unbuffer, this assumes the command only outputs to stderr.
+    """
+    try:
+        cmd = cmdline_s[0]
+        args = list(cmdline_s[1:])
+        child = pexpect.spawn(command=cmd, args=args)
+        child.expect(pexpect.EOF)
+        child.wait()  # this is needed to avoid a race condition on child.exitstatus
+
+        # child.exitstatus should never be None if child.wait() finished, but
+        # mypy doesn't know that.  Will treat that situation as a command error.
+        returncode = -1 if child.exitstatus is None else child.exitstatus
```

Comment:
if I understand correctly this line is just to make mypy happy?
Then I think I'd prefer something like `assert child.exitstatus is not None`